subsys/nfc/ndef/msg_parser.c

```diff
@@ -3,16 +3,20 @@
  *
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
+#include <kernel.h>
 #include <logging/log.h>
 #include "msg_parser_local.h"
 
 LOG_MODULE_REGISTER(nfc_ndef_parser, CONFIG_NFC_NDEF_PARSER_LOG_LEVEL);
 
-int nfc_ndef_msg_parse(const uint8_t *result_buf,
+int nfc_ndef_msg_parse(uint8_t *result_buf,
 		       uint32_t *result_buf_len,
 		       const uint8_t *raw_data,
 		       uint32_t *raw_data_len)
 {
+	__ASSERT((POINTER_TO_UINT(result_buf) % 4) == 0,
```

Comment:
Just a question. Shouldn't we also check if result_buf_len value is divisible by 4 ?

Reply:
It should be divisible by 4 as the output buffer should essentially contain just pointers. But I think there already are sufficient memory checks in the existing code:
https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/nfc/ndef/msg_parser_local.c#L65
https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/nfc/ndef/msg.c#L90